src/core/main.c

```diff
@@ -1523,7 +1523,7 @@ int main(int argc, char *argv[]) {
                 /* Try to figure out if we can use colors with the console. No
                  * need to do that for user instances since they never log
                  * into the console. */
-                log_show_color(colors_enabled());
+                log_show_color(colors_enabled_no_tty(true));
                 make_null_stdio();
```

Comment:
so, what about this. make colors_enabled() smart, so that it checks getpid() == 1, and if so doesn't care about the on_tty() thing? I mean, PID 1 is special in that regard: it generally outputs to the console, but doesn't keep it open all the time.